### ISSUE TITLE
Azure Storage using http

### ISSUE THREAD

**Issue by drbolsen** [pkgcloud/pkgcloud#526]
Just stumbled across a piece of code that handles Azure Storage operations that is actually using plain **http** instead of secure **https**:

```
  return urlJoin('http://' + this.azureKeys.storageAccount + '.' + this.serversUrl + '/', (typeof options === 'string'
```

I wonder if is there any particular reason for that? From the code it seems the protocol is hardcoded but I would love to see a configuration option which allows to select either protocol, based on one's requirements.

P.S. I did a quick dirty check by replacing http to https - it seems working just fine.

**Comment by drbolsen** [comment]
Is this project still alive? Wonder if we are better off forking and fixing it ourselves?

**Comment by BobDickinson** [comment]
Yes, this project is alive.  If you want to fix this yourself and submit a PR there's a good chance that someone will review it and merge it.  That's how I became a maintainer here (by fixing the stuff I cared about and submitting PRs).

**Comment by drbolsen** [comment]
Sure, no worries.

**Comment by sebastianfelipe** [comment]
I'm gonna push a PR in few minutes under the name of this issue.

**Comment by sebastianfelipe** [comment]
This is the solution. Is just use the this.protocol value already set, by default is https:// but you can change it in your configs as "protocol" #648

**Comment by rosskukulinski** [comment]
Fixed in #648
